### PR TITLE
Add validation of max temp. during dynamic adjustment

### PR DESCRIPTION
**Pull request by xylar**
This will hopefully catch issues like #308

### CODE REVIEW

**Comment by xylar** [comment]
## Testing

I'm testing dynamic adjustment on Anvil and Intel with this test.
- [ ] EC30to60: this fails as expected
```
ValueError: Max of temperatureMax > allowed threshold: 54.22333235731241 > 33.0 in /lcrc/group/e3sm/ac.xylar/compass_1.0/anvil/test_20220321/dyn_adj_ec_validate_temp_redi_min_6/ocean/global_ocean/EC30to60/PHC/dynamic_adjustment/damped_adjustment_2/analysis_members/globalStats.0001-01-11_00.00.00.nc
```
- [ ] ECwISC30to60
- [ ] QU240
- [ ] QUwISC240
- [ ] SOwISC12to60
- [ ] WC14

File: compass/ocean/tests/global_ocean/global_ocean.cfg [hunk 1]
```diff
@@ -71,6 +71,13 @@ email = autodetect
 pull_request = <<<Missing>>>
 
 
+# config options related to dynamic adjustment
+[dynamic_adjustment]
+
+# the maximum allowed value of temperatureMax in global statistics
+temperature_max = 33.0
```
Comment (xylar) [inline compass/ocean/tests/global_ocean/global_ocean.cfg:78]: I need to make sure this threshold is reasonable.

**Comment by xylar** [comment]
@mark-petersen, this is ready for your review.  Is 33.0 C a good threshold in you opinion?

**Review by mark-petersen — APPROVED** [review]
@xylar this is a great idea! Thanks for implementing it. Yes, I think 33C is an appropriate value. We can be conservative here, as it will tell us if the max is increasing at all.

Approving based on this catching increasing temperatures in the tests above.

**Comment by xylar** [comment]
Thanks again @mark-petersen!

**Pull request merged**